apps/dashboard/test/integration/projects_controller_test.rb

```diff
@@ -48,6 +48,7 @@ def setup
         get projects_path
         assert_response :success
 
+        # no projects directory created 
         assert Dir.exist?("#{dir}/projects")
     end
```

Comment:
This test was previously passing but fails now. I can't understand why it fails though. The `index` action should trigger `Project.all` which itself then calls `Project.dataroot` and uses a `tap` to then `join('projects')` to the `dataroot` but it doesn't look like the `join(projects)` ever occurs. 

My lack of understanding of `tap` may be the issue here. Does this actually create a directory?

Reply:
Looks like you changed `Project#dataroot`. Left a comment above.